flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/typeutils/LinkedListSerializer.java

```diff
@@ -186,12 +267,28 @@ public int getCurrentOuterSnapshotVersion() {
             return CURRENT_VERSION;
         }
 
+        @Override
+        protected void readOuterSnapshot(
+                int readOuterSnapshotVersion, DataInputView in, ClassLoader userCodeClassLoader) {
+            readVersion = readOuterSnapshotVersion;
+        }
+
+        @Override
+        protected OuterSchemaCompatibility resolveOuterSchemaCompatibility(
+                LinkedListSerializer<T> newSerializer) {
+            if (readVersion <= LAST_NULL_NOT_CHECKED_VERSION) {
+                return OuterSchemaCompatibility.COMPATIBLE_AFTER_MIGRATION;
```

Comment:
We could consider `NullableSerializer.checkIfNullSupported` too.

Reply:
Or you can use `MaskUtils` and ignore `checkIfNullSupported`. This way will not waste much space.